src/main/java/net/yogstation/yogbot/listeners/commands/BoundCredentialsCommand.kt

```diff
@@ -0,0 +1,165 @@
+package net.yogstation.yogbot.listeners.commands
+
+import discord4j.core.event.domain.message.MessageCreateEvent
+import net.yogstation.yogbot.DatabaseManager
+import net.yogstation.yogbot.Yogbot
+import net.yogstation.yogbot.config.DiscordConfig
+import net.yogstation.yogbot.permissions.PermissionsManager
+import net.yogstation.yogbot.util.DiscordUtil
+import net.yogstation.yogbot.util.StringUtils
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+import java.sql.SQLException
+import java.sql.Statement
+import java.sql.Types
+
+@Component
+class BoundCredentialsCommand(discordConfig: DiscordConfig,
+							  permissions: PermissionsManager,
+							  private val databaseManager: DatabaseManager
+) : PermissionsCommand(discordConfig, permissions) {
+	override val name = "bcreds"
+	override val requiredPermissions = "bcreds"
+	override val description = "Manages bound credentials"
+
+	override fun doCommand(event: MessageCreateEvent): Mono<*> {
+		val args = event.message.content.split(" ").toTypedArray()
+		return if (args.size < 2) {
+			DiscordUtil.reply(event, "Usage is `${args[0]} <add|search|delete>`")
+		} else when (args[1]) {
+			"add" -> addBoundCredential(event, args)
+			"search" -> findBoundCredential(event, args)
+			"delete" -> removeBoundCredential(event, args)
+			else -> DiscordUtil.reply(event, "Unknown subcommand: `${args[1]}`")
+		}
+	}
+
+	private fun addBoundCredential(event: MessageCreateEvent, args: Array<String>): Mono<*> {
+		if (args.size < 5) {
+			return DiscordUtil.reply(event, "Usage is `${args[0]} ${args[1]} <ckey> <cid> <ip>`\n" +
+				"Use `null` to ignore cid or ip. IP can be ascii or decimal")
+		}
+		val ipField = if(args[4].toIntOrNull() == null) "INET_ATON(?)" else "?"
+		databaseManager.byondDbConnection.use {conn -> conn.prepareStatement("""
+			INSERT INTO ${databaseManager.prefix("bound_credentials")} (`ckey`, `computerid`, `ip`) VALUES (?, ?, $ipField)
+		""".trimIndent(), Statement.RETURN_GENERATED_KEYS).use { preparedStatement ->
+			preparedStatement.setString(1, StringUtils.ckeyIze(args[2]))
+			if(args[3] == "null") preparedStatement.setNull(2, Types.VARCHAR) else preparedStatement.setString(2, args[3])
+			if(args[4] == "null") preparedStatement.setNull(3, Types.VARCHAR) else preparedStatement.setString(3, args[4])
+			try {
+				val rows = preparedStatement.executeUpdate()
+				if(rows == 0) throw SQLException("No rows inserted")
+			} catch (e: SQLException) {
+				return DiscordUtil.reply(event, "An error has occurred: ${e.message}")
+			}
+			preparedStatement.generatedKeys.use {
+				if(!it.next()) return DiscordUtil.reply(event, "No ID returned")
+				val bindings = (if(args[3] == "null") 0 else 1) + (if(args[4] == "null") 0 else 2)
+				return DiscordUtil.reply(event,
+					"Created binding ${it.getLong(1)}: ckey `${args[2]}` is now bound to " +
+						"${if(bindings and 1 == 1) "cid `${args[3]}`" else ""}${if(bindings == 3) " and " else ""}" +
```

Comment:
`1 == 1` ?

Reply:
That is a bitwise and, I'll add parenthesis for clarity